--- a/codemp/botlib/be_ai_move.cpp
+++ b/codemp/botlib/be_ai_move.cpp
@@ -2039,7 +2039,7 @@ bot_moveresult_t BotTravel_Elevator(bot_movestate_t *ms, aas_reachability_t *rea
 		botimport.Print(PRT_MESSAGE, "bot on elevator\n");
 #endif //DEBUG_ELEVATOR
 		//if vertically not too far from the end point
-		if (abs(ms->origin[2] - reach->end[2]) < sv_maxbarrier->value)
+		if (fabs(ms->origin[2] - reach->end[2]) < sv_maxbarrier->value)
 		{
 #ifdef DEBUG_ELEVATOR
 			botimport.Print(PRT_MESSAGE, "bot moving to end\n");
diff --git a/codemp/botlib/l_script.cpp b/codemp/botlib/l_script.cpp
index 7d772eb..acbad71 100644
