--- a/codemp/botlib/be_aas_entity.cpp
+++ b/codemp/botlib/be_aas_entity.cpp
@@ -373,9 +373,9 @@ int AAS_NearestEntity(vec3_t origin, int modelindex)
 		ent = &aasworld.entities[i];
 		if (ent->i.modelindex != modelindex) continue;
 		VectorSubtract(ent->i.origin, origin, dir);
-		if (abs(dir[0]) < 40)
+		if (fabs(dir[0]) < 40)
 		{
-			if (abs(dir[1]) < 40)
+			if (fabs(dir[1]) < 40)
 			{
 				dist = VectorLength(dir);
 				if (dist < bestdist)
diff --git a/codemp/botlib/be_aas_move.cpp b/codemp/botlib/be_aas_move.cpp
index 87e9770..f78a4b5 100644
