--- a/codemp/cgame/cg_draw.c
+++ b/codemp/cgame/cg_draw.c
@@ -4287,7 +4287,7 @@ void CG_DrawSiegeMessage( const char *str, int objectiveScreen )
 {
 //	if (!( trap_Key_GetCatcher() & KEYCATCH_UI ))
 	{
-		trap_OpenUIMenu(UIMENU_CLOSEALL);
+		/*trap_OpenUIMenu(UIMENU_CLOSEALL);
 		trap_Cvar_Set("cg_siegeMessage", str);
 		if (objectiveScreen)
 		{
@@ -4296,7 +4296,7 @@ void CG_DrawSiegeMessage( const char *str, int objectiveScreen )
 		else
 		{
 			trap_OpenUIMenu(UIMENU_SIEGEMESSAGE);
-		}
+		}*/
 	}
 }
 
diff --git a/codemp/cgame/cg_event.c b/codemp/cgame/cg_event.c
index 29cbbe4..15b5144 100644
