--- a/codemp/cgame/cg_event.c
+++ b/codemp/cgame/cg_event.c
@@ -2874,10 +2874,10 @@ void CG_EntityEvent( centity_t *cent, vec3_t position ) {
 
 			trap_Cvar_Set("ui_myteam", va("%i", es->bolt2));
 
-			if (!( trap_Key_GetCatcher() & KEYCATCH_UI ) && !es->bolt1)
+			/*if (!( trap_Key_GetCatcher() & KEYCATCH_UI ) && !es->bolt1)
 			{
 				trap_OpenUIMenu(UIMENU_PLAYERCONFIG);
-			}
+			}*/
 		}
 		break;
 
diff --git a/codemp/cgame/cg_public.h b/codemp/cgame/cg_public.h
index 44e85bb..0bafb6f 100644
