--- a/codemp/client/cl_keys.cpp
+++ b/codemp/client/cl_keys.cpp
@@ -2,7 +2,6 @@
 #include "qcommon/exe_headers.h"
 
 #include "client.h"
-#include "qcommon/stringed_ingame.h"
 /*
 
 key up events are sent even if in console mode
@@ -16,340 +15,135 @@ int			chat_playerNum;
 
 keyGlobals_t	kg;										
 
-// do NOT blithely change any of the key names (3rd field) here, since they have to match the key binds
-//	in the CFG files, they're also prepended with "KEYNAME_" when looking up StringEd references
-//
-keyname_t keynames[MAX_KEYS] =							
-{														
-	{ 0x00, 0x00, NULL, A_NULL, false									},					  
-	{ 0x01, 0x01, "SHIFT", A_SHIFT, false 								},					       
-	{ 0x02, 0x02, "CTRL", A_CTRL, false   								},					       
-	{ 0x03, 0x03, "ALT", A_ALT, false									},					       
-	{ 0x04, 0x04, "CAPSLOCK", A_CAPSLOCK, false							},					       
-	{ 0x05, 0x05, "KP_NUMLOCK", A_NUMLOCK, false						},					       
-	{ 0x06, 0x06, "SCROLLLOCK", A_SCROLLLOCK, false						},					       
-	{ 0x07, 0x07, "PAUSE", A_PAUSE, false								},					       
-	{ 0x08, 0x08, "BACKSPACE", A_BACKSPACE, false						},					       
-	{ 0x09, 0x09, "TAB", A_TAB, false									},					       
-	{ 0x0a, 0x0a, "ENTER", A_ENTER, false								},					       
-	{ 0x0b, 0x0b, "KP_PLUS", A_KP_PLUS, false							},					       
-	{ 0x0c, 0x0c, "KP_MINUS", A_KP_MINUS, false							},					       
-	{ 0x0d, 0x0d, "KP_ENTER", A_KP_ENTER, false							},					       
-	{ 0x0e, 0x0e, "KP_DEL", A_KP_PERIOD, false							},					       
-	{ 0x0f, 0x0f, NULL, A_PRINTSCREEN, false							},
-	{ 0x10, 0x10, "KP_INS", A_KP_0, false								},					       
-	{ 0x11, 0x11, "KP_END", A_KP_1, false								},					       
-	{ 0x12, 0x12, "KP_DOWNARROW", A_KP_2, false							},					       
-	{ 0x13, 0x13, "KP_PGDN", A_KP_3, false								},					       
-	{ 0x14, 0x14, "KP_LEFTARROW", A_KP_4, false							},					       
-	{ 0x15, 0x15, "KP_5", A_KP_5, false									},					       
-	{ 0x16, 0x16, "KP_RIGHTARROW", A_KP_6, false						},					       
-	{ 0x17, 0x17, "KP_HOME", A_KP_7, false								},					       
-	{ 0x18, 0x18, "KP_UPARROW", A_KP_8, false							},					       
-	{ 0x19, 0x19, "KP_PGUP", A_KP_9, false								},					       
-	{ 0x1a, 0x1a, "CONSOLE", A_CONSOLE, false 							},					       
-	{ 0x1b, 0x1b, "ESCAPE", A_ESCAPE, false								},					       
-	{ 0x1c, 0x1c, "F1", A_F1, true										},					       
-	{ 0x1d, 0x1d, "F2", A_F2, true										},					       
-	{ 0x1e, 0x1e, "F3", A_F3, true										},					       
-	{ 0x1f, 0x1f, "F4", A_F4, true										},					       
-			    													        
-	{ 0x20, 0x20, "SPACE", A_SPACE, false								},
-	{ (word)'!', (word)'!', NULL, A_PLING, false		  				},
-	{ (word)'"', (word)'"', NULL, A_DOUBLE_QUOTE, false  				},
-	{ (word)'#', (word)'#', NULL, A_HASH, false		  					},
-	{ (word)'$', (word)'$', NULL, A_STRING, false						},
-	{ (word)'%', (word)'%', NULL, A_PERCENT, false						},
-	{ (word)'&', (word)'&', NULL, A_AND, false							},
-	{ 0x27, 0x27, NULL, A_SINGLE_QUOTE, false							},
-	{ (word)'(', (word)'(', NULL, A_OPEN_BRACKET, false					},
-	{ (word)')', (word)')', NULL, A_CLOSE_BRACKET, false				},
-	{ (word)'*', (word)'*', NULL, A_STAR, false							},
-	{ (word)'+', (word)'+', NULL, A_PLUS, false							},
-	{ (word)',', (word)',', NULL, A_COMMA, false						},
-	{ (word)'-', (word)'-', NULL, A_MINUS, false						},
-	{ (word)'.', (word)'.', NULL, A_PERIOD, false						},
-	{ (word)'/', (word)'/', NULL, A_FORWARD_SLASH, false				},
-	{ (word)'0', (word)'0', NULL, A_0, false							},
-	{ (word)'1', (word)'1', NULL, A_1, false							},
-	{ (word)'2', (word)'2', NULL, A_2, false							},
-	{ (word)'3', (word)'3', NULL, A_3, false							},
-	{ (word)'4', (word)'4', NULL, A_4, false							},
-	{ (word)'5', (word)'5', NULL, A_5, false							},
-	{ (word)'6', (word)'6', NULL, A_6, false							},
-	{ (word)'7', (word)'7', NULL, A_7, false							},
-	{ (word)'8', (word)'8', NULL, A_8, false							},
-	{ (word)'9', (word)'9', NULL, A_9, false							},
-	{ (word)':', (word)':', NULL, A_COLON, false						},
-	{ (word)';', (word)';', "SEMICOLON", A_SEMICOLON, false				},
-	{ (word)'<', (word)'<', NULL, A_LESSTHAN, false						},
-	{ (word)'=', (word)'=', NULL, A_EQUALS, false						},
-	{ (word)'>', (word)'>', NULL, A_GREATERTHAN, false					},
-	{ (word)'?', (word)'?', NULL, A_QUESTION, false						},
-			    													
-	{ (word)'@', (word)'@', NULL, A_AT, false							},	   	            
-	{ (word)'A', (word)'a', NULL, A_CAP_A, false						},		            
-	{ (word)'B', (word)'b', NULL, A_CAP_B, false						},		            
-	{ (word)'C', (word)'c', NULL, A_CAP_C, false						},		            
-	{ (word)'D', (word)'d', NULL, A_CAP_D, false						},		            
-	{ (word)'E', (word)'e', NULL, A_CAP_E, false						},		            
-	{ (word)'F', (word)'f', NULL, A_CAP_F, false						},		            
-	{ (word)'G', (word)'g', NULL, A_CAP_G, false						},		            
-	{ (word)'H', (word)'h', NULL, A_CAP_H, false						},		            
-	{ (word)'I', (word)'i', NULL, A_CAP_I, false						},		            
-	{ (word)'J', (word)'j', NULL, A_CAP_J, false						},		            
-	{ (word)'K', (word)'k', NULL, A_CAP_K, false						},		            
-	{ (word)'L', (word)'l', NULL, A_CAP_L, false						},		            
-	{ (word)'M', (word)'m', NULL, A_CAP_M, false						},		            
-	{ (word)'N', (word)'n', NULL, A_CAP_N, false						},		            
-	{ (word)'O', (word)'o', NULL, A_CAP_O, false						},		            
-	{ (word)'P', (word)'p', NULL, A_CAP_P, false						},		            
-	{ (word)'Q', (word)'q', NULL, A_CAP_Q, false						},		            
-	{ (word)'R', (word)'r', NULL, A_CAP_R, false						},		            
-	{ (word)'S', (word)'s', NULL, A_CAP_S, false						},		            
-	{ (word)'T', (word)'t', NULL, A_CAP_T, false						},		            
-	{ (word)'U', (word)'u', NULL, A_CAP_U, false						},		            
-	{ (word)'V', (word)'v', NULL, A_CAP_V, false						},		            
-	{ (word)'W', (word)'w', NULL, A_CAP_W, false						},		            
-	{ (word)'X', (word)'x', NULL, A_CAP_X, false						},		            
-	{ (word)'Y', (word)'y', NULL, A_CAP_Y, false						},		            
-	{ (word)'Z', (word)'z', NULL, A_CAP_Z, false						},		            
-	{ (word)'[', (word)'[', NULL, A_OPEN_SQUARE, false					},					    
-	{ 0x5c, 0x5c, NULL, A_BACKSLASH, false								},					    
-	{ (word)']', (word)']', NULL, A_CLOSE_SQUARE, false 				},					    
-	{ (word)'^', (word)'^', NULL, A_CARET, false		 				},					    
-	{ (word)'_', (word)'_', NULL, A_UNDERSCORE, false					},					    
-			    													   
-	{ 0x60, 0x60, NULL, A_LEFT_SINGLE_QUOTE, false						},
-	{ (word)'A', (word)'a', NULL, A_LOW_A, false						},           
-	{ (word)'B', (word)'b', NULL, A_LOW_B, false						},           
-	{ (word)'C', (word)'c', NULL, A_LOW_C, false						},           
-	{ (word)'D', (word)'d', NULL, A_LOW_D, false						},           
-	{ (word)'E', (word)'e', NULL, A_LOW_E, false						},           
-	{ (word)'F', (word)'f', NULL, A_LOW_F, false						},           
-	{ (word)'G', (word)'g', NULL, A_LOW_G, false						},           
-	{ (word)'H', (word)'h', NULL, A_LOW_H, false						},           
-	{ (word)'I', (word)'i', NULL, A_LOW_I, false						},           
-	{ (word)'J', (word)'j', NULL, A_LOW_J, false						},           
-	{ (word)'K', (word)'k', NULL, A_LOW_K, false						},           
-	{ (word)'L', (word)'l', NULL, A_LOW_L, false						},           
-	{ (word)'M', (word)'m', NULL, A_LOW_M, false						},           
-	{ (word)'N', (word)'n', NULL, A_LOW_N, false						},           
-	{ (word)'O', (word)'o', NULL, A_LOW_O, false						},           
-	{ (word)'P', (word)'p', NULL, A_LOW_P, false						},           
-	{ (word)'Q', (word)'q', NULL, A_LOW_Q, false						},           
-	{ (word)'R', (word)'r', NULL, A_LOW_R, false						},           
-	{ (word)'S', (word)'s', NULL, A_LOW_S, false						},           
-	{ (word)'T', (word)'t', NULL, A_LOW_T, false						},           
-	{ (word)'U', (word)'u', NULL, A_LOW_U, false						},           
-	{ (word)'V', (word)'v', NULL, A_LOW_V, false						},           
-	{ (word)'W', (word)'w', NULL, A_LOW_W, false						},           
-	{ (word)'X', (word)'x', NULL, A_LOW_X, false						},           
-	{ (word)'Y', (word)'y', NULL, A_LOW_Y, false						},           
-	{ (word)'Z', (word)'z', NULL, A_LOW_Z, false						},           
-	{ (word)'{', (word)'{', NULL, A_OPEN_BRACE, false					},
-	{ (word)'|', (word)'|', NULL, A_BAR, false							},
-	{ (word)'}', (word)'}', NULL, A_CLOSE_BRACE, false					},
-	{ (word)'~', (word)'~', NULL, A_TILDE, false						},
-	{ 0x7f, 0x7f, "DEL", A_DELETE, false								},
-			    													               
-	{ 0x80, 0x80, "EURO", A_EURO, false  								},
-	{ 0x81, 0x81, "SHIFT", A_SHIFT2, false								},
-	{ 0x82, 0x82, "CTRL", A_CTRL2, false								},
-	{ 0x83, 0x83, "ALT", A_ALT2, false									},
-	{ 0x84, 0x84, "F5", A_F5, true										},
-	{ 0x85, 0x85, "F6", A_F6, true										},
-	{ 0x86, 0x86, "F7", A_F7, true										},
-	{ 0x87, 0x87, "F8", A_F8, true										},
-	{ 0x88, 0x88, "CIRCUMFLEX", A_CIRCUMFLEX, false  					},
-	{ 0x89, 0x89, "MWHEELUP", A_MWHEELUP, false							},
-	{ 0x8a, 0x9a, NULL, A_CAP_SCARON, false								},	// ******
-	{ 0x8b, 0x8b, "MWHEELDOWN", A_MWHEELDOWN, false						},
-	{ 0x8c, 0x9c, NULL, A_CAP_OE, false									},	// ******
-	{ 0x8d, 0x8d, "MOUSE1", A_MOUSE1, false								},
-	{ 0x8e, 0x8e, "MOUSE2", A_MOUSE2, false								},
-	{ 0x8f, 0x8f, "INS", A_INSERT, false								},
-	{ 0x90, 0x90, "HOME", A_HOME, false									},
-	{ 0x91, 0x91, "PGUP", A_PAGE_UP, false								},
-	{ 0x92, 0x92, NULL, A_RIGHT_SINGLE_QUOTE, false						},
-	{ 0x93, 0x93, NULL, A_LEFT_DOUBLE_QUOTE, false						},
-	{ 0x94, 0x94, NULL, A_RIGHT_DOUBLE_QUOTE, false						},
-	{ 0x95, 0x95, "F9", A_F9, true										},
-	{ 0x96, 0x96, "F10", A_F10, true									},
-	{ 0x97, 0x97, "F11", A_F11, true									},
-	{ 0x98, 0x98, "F12", A_F12, true									},
-	{ 0x99, 0x99, NULL, A_TRADEMARK, false								},
-	{ 0x8a, 0x9a, NULL, A_LOW_SCARON, false								},	// ******
-	{ 0x9b, 0x9b, "SHIFT_ENTER", A_ENTER, false							},
-	{ 0x8c, 0x9c, NULL, A_LOW_OE, false									},	// ******
-	{ 0x9d, 0x9d, "END", A_END, false									},
-	{ 0x9e, 0x9e, "PGDN", A_PAGE_DOWN, false							},
-	{ 0x9f, 0xff, NULL, A_CAP_YDIERESIS, false							},	// ******
-			    													
-	{ 0xa0, 0,	  "SHIFT_SPACE", A_SPACE, false							},
-	{ 0xa1, 0xa1, NULL, A_EXCLAMDOWN, false								},	// upside down '!' - undisplayable
-	{ (word)(byte)'¢', (word)(byte)'¢', NULL, A_CENT, false	  			},
-	{ (word)(byte)'£', (word)(byte)'£', NULL, A_POUND, false	  		},
-	{ 0xa4, 0,    "SHIFT_KP_ENTER", A_KP_ENTER, false					},
-	{ (word)(byte)'¥', (word)(byte)'¥', NULL, A_YEN, false		  		},
-	{ 0xa6, 0xa6, "MOUSE3", A_MOUSE3, false								},
-	{ 0xa7, 0xa7, "MOUSE4", A_MOUSE4, false								},
-	{ 0xa8, 0xa8, "MOUSE5", A_MOUSE5, false								},
-	{ (word)(byte)'©', (word)(byte)'©', NULL, A_COPYRIGHT, false 		},
-	{ 0xaa, 0xaa, "UPARROW", A_CURSOR_UP, false							},
-	{ 0xab, 0xab, "DOWNARROW", A_CURSOR_DOWN, false						},
-	{ 0xac, 0xac, "LEFTARROW", A_CURSOR_LEFT, false						},
-	{ 0xad, 0xad, "RIGHTARROW", A_CURSOR_RIGHT, false					},
-	{ (word)(byte)'®', (word)(byte)'®', NULL, A_REGISTERED, false		},
-	{ 0xaf, 0,	  NULL, A_UNDEFINED_7, false							},
-	{ 0xb0, 0,	  NULL, A_UNDEFINED_8, false							},
-	{ 0xb1, 0,	  NULL, A_UNDEFINED_9, false							},
-	{ 0xb2, 0,	  NULL, A_UNDEFINED_10, false							},
-	{ 0xb3, 0,	  NULL, A_UNDEFINED_11, false							},
-	{ 0xb4, 0,	  NULL, A_UNDEFINED_12, false							},
-	{ 0xb5, 0,	  NULL, A_UNDEFINED_13, false							},
-	{ 0xb6, 0,	  NULL, A_UNDEFINED_14, false							},
-	{ 0xb7, 0,	  NULL, A_UNDEFINED_15, false							},
-	{ 0xb8, 0,	  NULL, A_UNDEFINED_16, false							},
-	{ 0xb9, 0,	  NULL, A_UNDEFINED_17, false							},
-	{ 0xba, 0,	  NULL, A_UNDEFINED_18, false							},
-	{ 0xbb, 0,	  NULL, A_UNDEFINED_19, false							},
-	{ 0xbc, 0,	  NULL, A_UNDEFINED_20, false							},
-	{ 0xbd, 0,	  NULL, A_UNDEFINED_21, false							},
-	{ 0xbe, 0,	  NULL, A_UNDEFINED_22, false							},
-	{ (word)(byte)'¿', (word)(byte)'¿', NULL, A_QUESTION_DOWN, false	},
-			    		                       
-	{ (word)(byte)'À', (word)(byte)'à', NULL, A_CAP_AGRAVE, false		},
-	{ (word)(byte)'Á', (word)(byte)'á', NULL, A_CAP_AACUTE, false		},
-	{ (word)(byte)'Â', (word)(byte)'â', NULL, A_CAP_ACIRCUMFLEX, false	},
-	{ (word)(byte)'Ã', (word)(byte)'ã', NULL, A_CAP_ATILDE, false		},
-	{ (word)(byte)'Ä', (word)(byte)'ä', NULL, A_CAP_ADIERESIS, false	},
-	{ (word)(byte)'Å', (word)(byte)'å', NULL, A_CAP_ARING, false		},
-	{ (word)(byte)'Æ', (word)(byte)'æ', NULL, A_CAP_AE, false			},
-	{ (word)(byte)'Ç', (word)(byte)'ç', NULL, A_CAP_CCEDILLA, false		},
-	{ (word)(byte)'È', (word)(byte)'è', NULL, A_CAP_EGRAVE, false		},
-	{ (word)(byte)'É', (word)(byte)'é', NULL, A_CAP_EACUTE, false		},
-	{ (word)(byte)'Ê', (word)(byte)'ê', NULL, A_CAP_ECIRCUMFLEX, false	},
-	{ (word)(byte)'Ë', (word)(byte)'ë', NULL, A_CAP_EDIERESIS, false	},
-	{ (word)(byte)'Ì', (word)(byte)'ì', NULL, A_CAP_IGRAVE, false		},
-	{ (word)(byte)'Í', (word)(byte)'í', NULL, A_CAP_IACUTE, false		},
-	{ (word)(byte)'Î', (word)(byte)'î', NULL, A_CAP_ICIRCUMFLEX, false	},
-	{ (word)(byte)'Ï', (word)(byte)'ï', NULL, A_CAP_IDIERESIS, false	},
-	{ (word)(byte)'Ð', (word)(byte)'ð', NULL, A_CAP_ETH, false			},
-	{ (word)(byte)'Ñ', (word)(byte)'ñ', NULL, A_CAP_NTILDE, false		},
-	{ (word)(byte)'Ò', (word)(byte)'ò', NULL, A_CAP_OGRAVE, false		},
-	{ (word)(byte)'Ó', (word)(byte)'ó', NULL, A_CAP_OACUTE, false		},
-	{ (word)(byte)'Ô', (word)(byte)'ô', NULL, A_CAP_OCIRCUMFLEX, false	},
-	{ (word)(byte)'Õ', (word)(byte)'õ', NULL, A_CAP_OTILDE, false		},
-	{ (word)(byte)'Ö', (word)(byte)'ö', NULL, A_CAP_ODIERESIS, false	},
-	{ (word)(byte)'×', (word)(byte)'×', "KP_STAR", A_MULTIPLY, false 	},
-	{ (word)(byte)'Ø', (word)(byte)'ø', NULL, A_CAP_OSLASH, false		},  
-	{ (word)(byte)'Ù', (word)(byte)'ù', NULL, A_CAP_UGRAVE, false		},  
-	{ (word)(byte)'Ú', (word)(byte)'ú', NULL, A_CAP_UACUTE, false		},  
-	{ (word)(byte)'Û', (word)(byte)'û', NULL, A_CAP_UCIRCUMFLEX, false	},  
-	{ (word)(byte)'Ü', (word)(byte)'ü', NULL, A_CAP_UDIERESIS, false	},  
-	{ (word)(byte)'Ý', (word)(byte)'ý', NULL, A_CAP_YACUTE, false		},  
-	{ (word)(byte)'Þ', (word)(byte)'þ', NULL, A_CAP_THORN, false		},  
-	{ (word)(byte)'ß', (word)(byte)'ß', NULL, A_GERMANDBLS, false 		},
-			    		                       
-	{ (word)(byte)'À', (word)(byte)'à', NULL, A_LOW_AGRAVE, false		},		 
-	{ (word)(byte)'Á', (word)(byte)'á', NULL, A_LOW_AACUTE, false		},		 
-	{ (word)(byte)'Â', (word)(byte)'â', NULL, A_LOW_ACIRCUMFLEX, false	},		 
-	{ (word)(byte)'Ã', (word)(byte)'ã', NULL, A_LOW_ATILDE, false		},		 
-	{ (word)(byte)'Ä', (word)(byte)'ä', NULL, A_LOW_ADIERESIS, false	},		 
-	{ (word)(byte)'Å', (word)(byte)'å', NULL, A_LOW_ARING, false		},		 
-	{ (word)(byte)'Æ', (word)(byte)'æ', NULL, A_LOW_AE, false			},		 
-	{ (word)(byte)'Ç', (word)(byte)'ç', NULL, A_LOW_CCEDILLA, false		},		 
-	{ (word)(byte)'È', (word)(byte)'è', NULL, A_LOW_EGRAVE, false		},		 
-	{ (word)(byte)'É', (word)(byte)'é', NULL, A_LOW_EACUTE, false		},		 
-	{ (word)(byte)'Ê', (word)(byte)'ê', NULL, A_LOW_ECIRCUMFLEX, false	},		 
-	{ (word)(byte)'Ë', (word)(byte)'ë', NULL, A_LOW_EDIERESIS, false	},		 
-	{ (word)(byte)'Ì', (word)(byte)'ì', NULL, A_LOW_IGRAVE, false		},		 
-	{ (word)(byte)'Í', (word)(byte)'í', NULL, A_LOW_IACUTE, false		},		 
-	{ (word)(byte)'Î', (word)(byte)'î', NULL, A_LOW_ICIRCUMFLEX, false	},		 
-	{ (word)(byte)'Ï', (word)(byte)'ï', NULL, A_LOW_IDIERESIS, false	},		 
-	{ (word)(byte)'Ð', (word)(byte)'ð', NULL, A_LOW_ETH, false			},		 
-	{ (word)(byte)'Ñ', (word)(byte)'ñ', NULL, A_LOW_NTILDE, false		},		 
-	{ (word)(byte)'Ò', (word)(byte)'ò', NULL, A_LOW_OGRAVE, false		},		 
-	{ (word)(byte)'Ó', (word)(byte)'ó', NULL, A_LOW_OACUTE, false		},		 
-	{ (word)(byte)'Ô', (word)(byte)'ô', NULL, A_LOW_OCIRCUMFLEX, false	},		 
-	{ (word)(byte)'Õ', (word)(byte)'õ', NULL, A_LOW_OTILDE, false		},		 
-	{ (word)(byte)'Ö', (word)(byte)'ö', NULL, A_LOW_ODIERESIS, false	},		 
-	{ (word)(byte)'÷', (word)(byte)'÷', "KP_SLASH", A_DIVIDE, false 	},		 
-	{ (word)(byte)'Ø', (word)(byte)'ø', NULL, A_LOW_OSLASH, false		},		 
-	{ (word)(byte)'Ù', (word)(byte)'ù', NULL, A_LOW_UGRAVE, false		},		 
-	{ (word)(byte)'Ú', (word)(byte)'ú', NULL, A_LOW_UACUTE, false		},		 
-	{ (word)(byte)'Û', (word)(byte)'û', NULL, A_LOW_UCIRCUMFLEX, false	},		 
-	{ (word)(byte)'Ü', (word)(byte)'ü', NULL, A_LOW_UDIERESIS, false	},		 
-	{ (word)(byte)'Ý', (word)(byte)'ý', NULL, A_LOW_YACUTE, false		},		 
-	{ (word)(byte)'Þ', (word)(byte)'þ', NULL, A_LOW_THORN, false		},		 
-	{ 0x9f, 0xff, NULL, A_LOW_YDIERESIS, false							},	// *******
-																		
-	{ 0x100, 0x100, "JOY0", A_JOY0, false								},
-	{ 0x101, 0x101, "JOY1", A_JOY1, false								},
-	{ 0x102, 0x102, "JOY2", A_JOY2, false								},
-	{ 0x103, 0x103, "JOY3", A_JOY3, false								},
-	{ 0x104, 0x104, "JOY4", A_JOY4, false								},
-	{ 0x105, 0x105, "JOY5", A_JOY5, false								},
-	{ 0x106, 0x106, "JOY6", A_JOY6, false								},
-	{ 0x107, 0x107, "JOY7", A_JOY7, false								},
-	{ 0x108, 0x108, "JOY8", A_JOY8, false								},
-	{ 0x109, 0x109, "JOY9", A_JOY9, false								},
-	{ 0x10a, 0x10a, "JOY10", A_JOY10, false								},
-	{ 0x10b, 0x10b, "JOY11", A_JOY11, false								},
-	{ 0x10c, 0x10c, "JOY12", A_JOY12, false								},
-	{ 0x10d, 0x10d, "JOY13", A_JOY13, false								},
-	{ 0x10e, 0x10e, "JOY14", A_JOY14, false								},
-	{ 0x10f, 0x10f, "JOY15", A_JOY15, false								},
-	{ 0x110, 0x110, "JOY16", A_JOY16, false								},
-	{ 0x111, 0x111, "JOY17", A_JOY17, false								},
-	{ 0x112, 0x112, "JOY18", A_JOY18, false								},
-	{ 0x113, 0x113, "JOY19", A_JOY19, false								},
-	{ 0x114, 0x114, "JOY20", A_JOY20, false								},
-	{ 0x115, 0x115, "JOY21", A_JOY21, false								},
-	{ 0x116, 0x116, "JOY22", A_JOY22, false								},
-	{ 0x117, 0x117, "JOY23", A_JOY23, false								},
-	{ 0x118, 0x118, "JOY24", A_JOY24, false								},
-	{ 0x119, 0x119, "JOY25", A_JOY25, false								},
-	{ 0x11a, 0x11a, "JOY26", A_JOY26, false								},
-	{ 0x11b, 0x11b, "JOY27", A_JOY27, false								},
-	{ 0x11c, 0x11c, "JOY28", A_JOY28, false								},
-	{ 0x11d, 0x11d, "JOY29", A_JOY29, false								},
-	{ 0x11e, 0x11e, "JOY30", A_JOY30, false								},
-	{ 0x11f, 0x11f, "JOY31", A_JOY31, false								},
-		    	    												
-	{ 0x120, 0x120, "AUX0", A_AUX0, false								},
-	{ 0x121, 0x121, "AUX1", A_AUX1, false								},
-	{ 0x122, 0x122, "AUX2", A_AUX2, false								},
-	{ 0x123, 0x123, "AUX3", A_AUX3, false								},
-	{ 0x124, 0x124, "AUX4", A_AUX4, false								},
-	{ 0x125, 0x125, "AUX5", A_AUX5, false								},
-	{ 0x126, 0x126, "AUX6", A_AUX6, false								},
-	{ 0x127, 0x127, "AUX7", A_AUX7, false								},
-	{ 0x128, 0x128, "AUX8", A_AUX8, false								},
-	{ 0x129, 0x129, "AUX9", A_AUX9, false								},
-	{ 0x12a, 0x12a, "AUX10", A_AUX10, false								},
-	{ 0x12b, 0x12b, "AUX11", A_AUX11, false								},
-	{ 0x12c, 0x12c, "AUX12", A_AUX12, false								},
-	{ 0x12d, 0x12d, "AUX13", A_AUX13, false								},
-	{ 0x12e, 0x12e, "AUX14", A_AUX14, false								},
-	{ 0x12f, 0x12f, "AUX15", A_AUX15, false								},
-	{ 0x130, 0x130, "AUX16", A_AUX16, false								},
-	{ 0x131, 0x131, "AUX17", A_AUX17, false								},
-	{ 0x132, 0x132, "AUX18", A_AUX18, false								},
-	{ 0x133, 0x133, "AUX19", A_AUX19, false								},
-	{ 0x134, 0x134, "AUX20", A_AUX20, false								},
-	{ 0x135, 0x135, "AUX21", A_AUX21, false								},
-	{ 0x136, 0x136, "AUX22", A_AUX22, false								},
-	{ 0x137, 0x137, "AUX23", A_AUX23, false								},
-	{ 0x138, 0x138, "AUX24", A_AUX24, false								},
-	{ 0x139, 0x139, "AUX25", A_AUX25, false								},
-	{ 0x13a, 0x13a, "AUX26", A_AUX26, false								},
-	{ 0x13b, 0x13b, "AUX27", A_AUX27, false								},
-	{ 0x13c, 0x13c, "AUX28", A_AUX28, false								},
-	{ 0x13d, 0x13d, "AUX29", A_AUX29, false								},
-	{ 0x13e, 0x13e, "AUX30", A_AUX30, false								},
-	{ 0x13f, 0x13f, "AUX31", A_AUX31, false								}
+// names not in this list can either be lowercase ascii, or '0xnn' hex sequences
+keyname_t keynames[K_LAST_KEY] =							
+{			
+	{"TAB", K_TAB},
+	{"ENTER", K_ENTER},
+	{"ESCAPE", K_ESCAPE},
+	{"SPACE", K_SPACE},
+
+	{"BACKSPACE", K_BACKSPACE},
+
+	{"COMMAND", K_COMMAND},
+	{"CAPSLOCK", K_CAPSLOCK},
+	{"POWER", K_POWER},
+	{"PAUSE", K_PAUSE},
+
+	{"UPARROW", K_UPARROW},
+	{"DOWNARROW", K_DOWNARROW},
+	{"LEFTARROW", K_LEFTARROW},
+	{"RIGHTARROW", K_RIGHTARROW},
+
+	{"ALT", K_ALT},
+	{"CTRL", K_CTRL},
+	{"SHIFT", K_SHIFT},
+	{"INS", K_INS},
+	{"DEL", K_DEL},
+	{"PGDN", K_PGDN},
+	{"PGUP", K_PGUP},
+	{"HOME", K_HOME},
+	{"END", K_END},
+	
+	{"F1", K_F1},
+	{"F2", K_F2},
+	{"F3", K_F3},
+	{"F4", K_F4},
+	{"F5", K_F5},
+	{"F6", K_F6},
+	{"F7", K_F7},
+	{"F8", K_F8},
+	{"F9", K_F9},
+	{"F10", K_F10},
+	{"F11", K_F11},
+	{"F12", K_F12},
+	{"F13", K_F13},
+	{"F14", K_F14},
+	{"F15", K_F15},
+
+	{"SCROLL", K_SCROLL},
+
+	{"KP_HOME", K_KP_HOME },
+	{"KP_UPARROW", K_KP_UPARROW },
+	{"KP_PGUP", K_KP_PGUP },
+	{"KP_LEFTARROW", K_KP_LEFTARROW },
+	{"KP_5", K_KP_5 },
+	{"KP_RIGHTARROW", K_KP_RIGHTARROW },
+	{"KP_END", K_KP_END },
+	{"KP_DOWNARROW", K_KP_DOWNARROW },
+	{"KP_PGDN", K_KP_PGDN },
+	{"KP_ENTER",  K_KP_ENTER },
+	{"KP_INS", K_KP_INS },
+	{"KP_DEL", K_KP_DEL },
+	{"KP_SLASH", K_KP_SLASH },
+	{"KP_MINUS", K_KP_MINUS },
+	{"KP_PLUS", K_KP_PLUS },
+	{"KP_NUMLOCK", K_KP_NUMLOCK },
+	{"KP_STAR", K_KP_STAR },
+	{"KP_EQUALS", K_KP_EQUALS },
+	
+	{"MOUSE1", K_MOUSE1},
+	{"MOUSE2", K_MOUSE2},
+	{"MOUSE3", K_MOUSE3},
+	{"MOUSE4", K_MOUSE4},
+	{"MOUSE5", K_MOUSE5},
+
+	{"MWHEELUP",    K_MWHEELUP },
+	{"MWHEELDOWN",  K_MWHEELDOWN },
+
+	{"JOY1", K_JOY1},
+	{"JOY2", K_JOY2},
+	{"JOY3", K_JOY3},
+	{"JOY4", K_JOY4},
+	{"JOY5", K_JOY5},
+	{"JOY6", K_JOY6},
+	{"JOY7", K_JOY7},
+	{"JOY8", K_JOY8},
+	{"JOY9", K_JOY9},
+	{"JOY10", K_JOY10},
+	{"JOY11", K_JOY11},
+	{"JOY12", K_JOY12},
+	{"JOY13", K_JOY13},
+	{"JOY14", K_JOY14},
+	{"JOY15", K_JOY15},
+	{"JOY16", K_JOY16},
+	{"JOY17", K_JOY17},
+	{"JOY18", K_JOY18},
+	{"JOY19", K_JOY19},
+	{"JOY20", K_JOY20},
+	{"JOY21", K_JOY21},
+	{"JOY22", K_JOY22},
+	{"JOY23", K_JOY23},
+	{"JOY24", K_JOY24},
+	{"JOY25", K_JOY25},
+	{"JOY26", K_JOY26},
+	{"JOY27", K_JOY27},
+	{"JOY28", K_JOY28},
+	{"JOY29", K_JOY29},
+	{"JOY30", K_JOY30},
+	{"JOY31", K_JOY31},
+	{"JOY32", K_JOY32},
+
+	{"AUX1", K_AUX1},
+	{"AUX2", K_AUX2},
+	{"AUX3", K_AUX3},
+	{"AUX4", K_AUX4},
+	{"AUX5", K_AUX5},
+	{"AUX6", K_AUX6},
+	{"AUX7", K_AUX7},
+	{"AUX8", K_AUX8},
+	{"AUX9", K_AUX9},
+	{"AUX10", K_AUX10},
+	{"AUX11", K_AUX11},
+	{"AUX12", K_AUX12},
+	{"AUX13", K_AUX13},
+	{"AUX14", K_AUX14},
+	{"AUX15", K_AUX15},
+	{"AUX16", K_AUX16},
+
+	{"SEMICOLON", ';'},     // because a raw semicolon seperates commands
+
+	{NULL, 0}
 };
 
 
@@ -460,7 +254,7 @@ void Field_Draw( field_t *edit, int x, int y, int width, qboolean showCursor )
 
 void Field_BigDraw( field_t *edit, int x, int y, int width, qboolean showCursor ) 
 {
-	Field_VariableSizeDraw( edit, x, y, width, BIGCHAR_WIDTH, showCursor );
+	Field_VariableSizeDraw( edit, x, y, width, GIANTCHAR_HEIGHT/2, showCursor );
 }
 
 /*
@@ -501,14 +295,14 @@ void Field_KeyDownEvent( field_t *edit, int key ) {
 	int		len;
 
 	// shift-insert is paste
-	if ( ( ( key == A_INSERT ) || ( key == A_KP_0 ) ) && kg.keys[A_SHIFT].down ) {
+	if ( ( ( key == K_INS ) || ( key == K_KP_INS ) ) && kg.keys[K_SHIFT].down ) {
 		Field_Paste( edit );
 		return;
 	}
 
 	len = strlen( edit->buffer );
 
-	if ( key == A_DELETE ) {
+	if ( key == K_DEL ) {
 		if ( edit->cursor < len ) {
 			memmove( edit->buffer + edit->cursor, 
 				edit->buffer + edit->cursor + 1, len - edit->cursor );
@@ -516,7 +310,7 @@ void Field_KeyDownEvent( field_t *edit, int key ) {
 		return;
 	}
 
-	if ( key == A_CURSOR_RIGHT ) 
+	if ( key == K_RIGHTARROW ) 
 	{
 		if ( edit->cursor < len ) {
 			edit->cursor++;
@@ -529,7 +323,7 @@ void Field_KeyDownEvent( field_t *edit, int key ) {
 		return;
 	}
 
-	if ( key == A_CURSOR_LEFT ) 
+	if ( key == K_LEFTARROW ) 
 	{
 		if ( edit->cursor > 0 ) {
 			edit->cursor--;
@@ -541,19 +335,19 @@ void Field_KeyDownEvent( field_t *edit, int key ) {
 		return;
 	}
 
-	if ( key == A_HOME || ( keynames[key].lower == 'a' && kg.keys[A_CTRL].down ) ) 
+	if ( key == K_HOME || ( tolower(key) == 'a' && kg.keys[K_CTRL].down ) ) 
 	{
 		edit->cursor = 0;
 		return;
 	}
 
-	if ( key == A_END || ( keynames[key].lower == 'e' && kg.keys[A_CTRL].down ) ) 
+	if ( key == K_END || ( tolower(key) == 'e' && kg.keys[K_CTRL].down ) ) 
 	{
 		edit->cursor = len;
 		return;
 	}
 
-	if ( key == A_INSERT ) {
+	if ( key == K_INS ) {
 		kg.key_overstrikeMode = (qboolean)!kg.key_overstrikeMode;
 		return;
 	}
@@ -807,13 +601,13 @@ Handles history and console scrollback
 */
 void Console_Key (int key) {
 	// ctrl-L clears screen
-	if ( keynames[ key ].lower == 'l' && kg.keys[A_CTRL].down ) {
+	if ( tolower(key) == 'l' && kg.keys[K_CTRL].down ) {
 		Cbuf_AddText ("clear\n");
 		return;
 	}
 
 	// enter finishes the line
-	if ( key == A_ENTER || key == A_KP_ENTER ) {
+	if ( key == K_ENTER || key == K_KP_ENTER ) {
 		// if not in the game explicitly prepent a slash if needed
 		if ( cls.state != CA_ACTIVE && kg.g_consoleField.buffer[0] != '\\' 
 			&& kg.g_consoleField.buffer[0] != '/' ) {
@@ -832,29 +626,8 @@ void Console_Key (int key) {
 
 		// leading slash is an explicit command
 		if ( kg.g_consoleField.buffer[0] == '\\' || kg.g_consoleField.buffer[0] == '/' ) {
-			if (cgvm && cl.mSharedMemory)
-			{ //don't do this unless cgame is inited and shared memory is valid
-				const char *buf = (kg.g_consoleField.buffer+1);
-				TCGIncomingConsoleCommand *icc = (TCGIncomingConsoleCommand *)cl.mSharedMemory;
-
-				strcpy(icc->conCommand, buf);
-				
-				if (VM_Call(cgvm, CG_INCOMING_CONSOLE_COMMAND))
-				{ //rww - let mod authors filter client console messages so they can cut them off if they want.
-					Cbuf_AddText( kg.g_consoleField.buffer+1 );	// valid command
-					Cbuf_AddText ("\n");
-				}
-				else if (icc->conCommand[0])
-				{ //the vm call says to execute this command in place
-					Cbuf_AddText( icc->conCommand );
-					Cbuf_AddText ("\n");
-				}
-			}
-			else
-			{ //just exec it then
-				Cbuf_AddText( kg.g_consoleField.buffer+1 );	// valid command
-				Cbuf_AddText ("\n");
-			}
+			Cbuf_AddText( kg.g_consoleField.buffer+1 );	// valid command
+			Cbuf_AddText ("\n");
 		} else {
 			// other text will be chat messages
 			if ( !kg.g_consoleField.buffer[0] ) {
@@ -883,14 +656,14 @@ void Console_Key (int key) {
 
 	// command completion
 
-	if (key == A_TAB) {
+	if (key == K_TAB) {
 		CompleteCommand();
 		return;
 	}
 
 	// command history (ctrl-p ctrl-n for unix style)
 
-	if ( ( key == A_MWHEELUP && kg.keys[A_SHIFT].down ) || ( key == A_CURSOR_UP ) || ( key == A_KP_8 ) || ( ( keynames[ key ].lower == 'p' ) && kg.keys[A_CTRL].down ) ) 
+	if ( ( key == K_MWHEELUP && kg.keys[K_SHIFT].down ) || ( key == K_UPARROW ) || ( key == K_KP_UPARROW ) || ( ( tolower(key) == 'p' ) && kg.keys[K_CTRL].down ) ) 
 	{
 		if ( kg.nextHistoryLine - kg.historyLine < COMMAND_HISTORY && kg.historyLine > 0 ) 
 		{
@@ -900,7 +673,7 @@ void Console_Key (int key) {
 		return;
 	}
 
-	if ( ( key == A_MWHEELDOWN && kg.keys[A_SHIFT].down ) || ( key == A_CURSOR_DOWN ) || ( key == A_KP_2 ) || ( ( keynames[ key ].lower == 'n' ) && kg.keys[A_CTRL].down ) ) 
+	if ( ( key == K_MWHEELDOWN && kg.keys[K_SHIFT].down ) || ( key == K_DOWNARROW ) || ( key == K_KP_DOWNARROW ) || ( ( tolower(key) == 'n' ) && kg.keys[K_CTRL].down ) ) 
 	{
 		if (kg.historyLine == kg.nextHistoryLine)
 			return;
@@ -910,28 +683,28 @@ void Console_Key (int key) {
 	}
 
 	// console scrolling
-	if ( key == A_PAGE_UP ) {
+	if ( key == K_PGUP ) {
 		Con_PageUp();
 		return;
 	}
 
-	if ( key == A_PAGE_DOWN ) {
+	if ( key == K_PGDN ) {
 		Con_PageDown();
 		return;
 	}
 
-	if ( key == A_MWHEELUP ) {	//----(SA)	added some mousewheel functionality to the console
+	if ( key == K_MWHEELUP ) {	//----(SA)	added some mousewheel functionality to the console
 		Con_PageUp();
-		if(kg.keys[A_CTRL].down) {	// hold <ctrl> to accelerate scrolling
+		if(kg.keys[K_CTRL].down) {	// hold <ctrl> to accelerate scrolling
 			Con_PageUp();
 			Con_PageUp();
 		}
 		return;
 	}
 
-	if ( key == A_MWHEELDOWN ) {	//----(SA)	added some mousewheel functionality to the console
+	if ( key == K_MWHEELDOWN ) {	//----(SA)	added some mousewheel functionality to the console
 		Con_PageDown();
-		if(kg.keys[A_CTRL].down) {	// hold <ctrl> to accelerate scrolling
+		if(kg.keys[K_CTRL].down) {	// hold <ctrl> to accelerate scrolling
 			Con_PageDown();
 			Con_PageDown();
 		}
@@ -939,13 +712,13 @@ void Console_Key (int key) {
 	}
 
 	// ctrl-home = top of console
-	if ( key == A_HOME && kg.keys[A_CTRL].down ) {
+	if ( key == K_HOME && kg.keys[K_CTRL].down ) {
 		Con_Top();
 		return;
 	}
 
 	// ctrl-end = bottom of console
-	if ( key == A_END && kg.keys[A_CTRL].down ) {
+	if ( key == K_END && kg.keys[K_CTRL].down ) {
 		Con_Bottom();
 		return;
 	}
@@ -967,13 +740,13 @@ In game talk message
 void Message_Key( int key ) {
 	char	buffer[MAX_STRING_CHARS];
 
-	if (key == A_ESCAPE) {
+	if (key == K_ESCAPE) {
 		cls.keyCatchers &= ~KEYCATCH_MESSAGE;
 		Field_Clear( &chatField );
 		return;
 	}
 
-	if ( key == A_ENTER || key == A_KP_ENTER )
+	if ( key == K_ENTER || key == K_KP_ENTER )
 	{
 		if ( chatField.buffer[0] && cls.state == CA_ACTIVE ) {
 			if (chat_playerNum != -1 )
@@ -1016,7 +789,7 @@ qboolean Key_IsDown( int keynum ) {
 		return qfalse;
 	}
 
-	return kg.keys[ keynames[keynum].upper ].down;
+	return kg.keys[keynum].down;
 }
 
 
@@ -1033,7 +806,7 @@ to be configured even if they don't have defined names.
 ===================
 */
 int Key_StringToKeynum( char *str ) {
-	int			i;
+	keyname_t	*kn;
 	
 	if ( !str || !str[0] ) 
 	{
@@ -1042,15 +815,15 @@ int Key_StringToKeynum( char *str ) {
 	// If single char bind, presume ascii char bind
 	if ( !str[1] ) 
 	{
-		return keynames[ (unsigned char)str[0] ].upper;
+		return str[0];
 	}
 
 	// scan for a text match
-	for ( i = 0 ; i < MAX_KEYS ; i++ ) 
+	for ( kn=keynames ; kn->name ; kn++ ) 
 	{
-		if ( keynames[i].name && !stricmp( str, keynames[i].name ) )
+		if ( !stricmp( str, kn->name ) )
 		{
-			return keynames[i].keynum;
+			return kn->keynum;
 		}
 	}
 
@@ -1100,45 +873,13 @@ static const char *Key_KeynumValid( int keynum )
 	{
 		return "<KEY NOT FOUND>";
 	}
-	if ( keynum < 0 || keynum >= MAX_KEYS ) 
+	if ( keynum < 0 || keynum >= K_LAST_KEY ) 
 	{
 		return "<OUT OF RANGE>";
 	}
 	return NULL;
 }
 
-static const char *Key_KeyToName( int keynum )
-{
-	return keynames[keynum].name;
-}
-
-
-static const char *Key_KeyToAscii( int keynum )
-{
-	if(!keynames[keynum].lower)
-	{
-		return(NULL);
-	}
-	if(keynum == A_SPACE)
-	{
-		tinyString[0] = (char)A_SHIFT_SPACE;
-	}
-	else if(keynum == A_ENTER)
-	{
-		tinyString[0] = (char)A_SHIFT_ENTER;
-	}
-	else if(keynum == A_KP_ENTER)
-	{
-		tinyString[0] = (char)A_SHIFT_KP_ENTER;
-	}
-	else
-	{
-		tinyString[0] = keynames[keynum].upper;
-	}
-	tinyString[1] = 0;
-	return tinyString;
-}
-
 static const char *Key_KeyToHex( int keynum )
 {
 	int		i, j;
@@ -1155,32 +896,6 @@ static const char *Key_KeyToHex( int keynum )
 	return tinyString;
 }
 
-// Returns the ascii code of the keynum
-const char *Key_KeynumToAscii( int keynum ) 
-{
-	const char	*name;
-
-	name = Key_KeynumValid(keynum);
-
-	// check for printable ascii
-	if ( !name && keynum > 0 && keynum < 256 ) 
-	{
-		name = Key_KeyToAscii(keynum);
-	}
-	// Check for name (for JOYx and AUXx buttons)
-	if ( !name )
-	{
-		name = Key_KeyToName(keynum);
-	}
-	// Fallback to hex number
-	if ( !name )
-	{
-		name = Key_KeyToHex(keynum);
-	}
-	return name;
-}
-
-
 /*
 ===================
 Key_KeynumToString
@@ -1192,19 +907,27 @@ given keynum.
 // Returns a console/config file friendly name for the key
 const char *Key_KeynumToString( int keynum ) 
 {
+	keyname_t   *kn;
 	const char	*name;
 
 	name = Key_KeynumValid(keynum);
 
-	// Check for friendly name
-	if ( !name )
+	// check for printable ascii
+	if ( !name && keynum > 32 && keynum < 127) 
 	{
-		name = Key_KeyToName(keynum);
+		tinyString[0] = keynum;
+		tinyString[1] = 0;
+		return tinyString;
 	}
-	// check for printable ascii
-	if ( !name && keynum > 0 && keynum < 256) 
+	// Check for friendly name
+	if ( !name )
 	{
-		name = Key_KeyToAscii(keynum);
+		for ( kn=keynames ; kn->name ; kn++ ) {
+			if (keynum == kn->keynum) {
+				name = kn->name;
+				break;
+			}
+		}
 	}
 	// Fallback to hex number
 	if ( !name )
@@ -1227,15 +950,15 @@ void Key_SetBinding( int keynum, const char *binding ) {
 	}
 
 	// free old bindings
-	if ( kg.keys[ keynames[keynum].upper ].binding ) {
-		Z_Free( kg.keys[ keynames[keynum].upper ].binding );
-		kg.keys[ keynames[keynum].upper ].binding = NULL;
+	if ( kg.keys[keynum].binding ) {
+		Z_Free( kg.keys[keynum].binding );
+		kg.keys[keynum].binding = NULL;
 	}
 			
 	// allocate memory for new binding
 	if (binding)
 	{
-		kg.keys[ keynames[keynum].upper ].binding = CopyString( binding );
+		kg.keys[keynum].binding = CopyString( binding );
 	}
 
 	// consider this like modifying an archived cvar, so the
@@ -1310,7 +1033,7 @@ void Key_Unbindall_f (void)
 {
 	int		i;
 	
-	for (i = 0; i < MAX_KEYS ; i++)
+	for (i = 0; i < K_LAST_KEY ; i++)
 	{
 		if (kg.keys[i].binding)
 		{
@@ -1377,7 +1100,7 @@ void Key_WriteBindings( fileHandle_t f ) {
 	int		i;
 
 	FS_Printf (f, "unbindall\n" );
-	for (i=0 ; i<MAX_KEYS ; i++) {
+	for (i=0 ; i<K_LAST_KEY ; i++) {
 		if (kg.keys[i].binding && kg.keys[i].binding[0] ) {
 			const char *name = Key_KeynumToString(i);
 
@@ -1401,9 +1124,9 @@ Key_Bindlist_f
 void Key_Bindlist_f( void ) {
 	int		i;
 
-	for ( i = 0 ; i < MAX_KEYS ; i++ ) {
+	for ( i = 0 ; i < K_LAST_KEY ; i++ ) {
 		if ( kg.keys[i].binding && kg.keys[i].binding[0] ) {
-			Com_Printf( "Key : %s (%s) \"%s\"\n", Key_KeynumToAscii(i), Key_KeynumToString(i), kg.keys[i].binding );
+			Com_Printf( "Key : %s \"%s\"\n", Key_KeynumToString(i), kg.keys[i].binding );
 		}
 	}
 }
@@ -1477,11 +1200,11 @@ void CL_KeyEvent (int key, qboolean down, unsigned time) {
 	char	cmd[1024];
 
 	// update auto-repeat status and BUTTON_ANY status
-	kg.keys[ keynames[key].upper ].down = down;
+	kg.keys[key].down = down;
 	if (down)
 	{
-		kg.keys[ keynames[key].upper ].repeats++;
-		if ( kg.keys[ keynames[key].upper ].repeats == 1 && key != A_SCROLLLOCK && key != A_NUMLOCK && key != A_CAPSLOCK )
+		kg.keys[key].repeats++;
+		if ( kg.keys[key].repeats == 1 && key != K_KP_NUMLOCK && key != K_CAPSLOCK )
 		{
 			kg.anykeydown = qtrue;
 			kg.keyDownCount++;
@@ -1489,8 +1212,8 @@ void CL_KeyEvent (int key, qboolean down, unsigned time) {
 	}
 	else
 	{
-		kg.keys[ keynames[key].upper ].repeats = 0;
-		if( key != A_SCROLLLOCK && key != A_NUMLOCK && key != A_CAPSLOCK )
+		kg.keys[key].repeats = 0;
+		if( key != K_KP_NUMLOCK && key != K_CAPSLOCK )
 			kg.keyDownCount--;
 		if(kg.keyDownCount <= 0)
 		{
@@ -1500,7 +1223,7 @@ void CL_KeyEvent (int key, qboolean down, unsigned time) {
 	}
 
 	// console key is hardcoded, so the user can never unbind it
-	if (key == A_CONSOLE || ( kg.keys[A_SHIFT].down && key == A_ESCAPE ) ) {
+	if (key == K_CONSOLE || ( kg.keys[K_SHIFT].down && key == K_ESCAPE ) ) {
 		if (!down) {
 			return;
 		}
@@ -1509,18 +1232,18 @@ void CL_KeyEvent (int key, qboolean down, unsigned time) {
 	}
 
 	// kg.keys can still be used for bound actions
-	if ( down && /*( key < 128 || key == A_MOUSE1 ) && */
+	if ( down && ( key < 128 || key == K_MOUSE1 ) && 
 		( cls.state == CA_CINEMATIC ) &&
 		!cls.keyCatchers) {
 
 		if (Cvar_VariableValue ("com_cameraMode") == 0) {
 			Cvar_Set ("nextdemo","");
-			key = A_ESCAPE;
+			key = K_ESCAPE;
 		}
 	}
 
 	// escape is always handled special
-	if ( key == A_ESCAPE && down ) {
+	if ( key == K_ESCAPE && down ) {
 		if ( cls.keyCatchers & KEYCATCH_MESSAGE ) {
 			// clear message mode
 			Message_Key( key );
@@ -1557,7 +1280,7 @@ void CL_KeyEvent (int key, qboolean down, unsigned time) {
 	// an action started before a mode switch.
 	//
 	if (!down) {
-		kb = kg.keys[ keynames[key].upper ].binding;
+		kb = kg.keys[key].binding;
 
 		CL_AddKeyUpCommands( key, kb );
 
@@ -1588,7 +1311,7 @@ void CL_KeyEvent (int key, qboolean down, unsigned time) {
 		Console_Key( key );
 	} else {
 		// send the bound action
-		kb = kg.keys[ keynames[key].upper ].binding;
+		kb = kg.keys[key].binding;
 		if (kb)
 		{
 			if (kb[0] == '+') {	
@@ -1620,28 +1343,8 @@ void CL_KeyEvent (int key, qboolean down, unsigned time) {
 				}
 			} else {
 				// down-only command
-				if (cgvm && cl.mSharedMemory)
-				{ //don't do this unless cgame is inited and shared memory is valid
-					TCGIncomingConsoleCommand *icc = (TCGIncomingConsoleCommand *)cl.mSharedMemory;
-
-					strcpy(icc->conCommand, kb);
-					
-					if (VM_Call(cgvm, CG_INCOMING_CONSOLE_COMMAND))
-					{ //rww - let mod authors filter client console messages so they can cut them off if they want.
-						Cbuf_AddText (kb);
-						Cbuf_AddText ("\n");
-					}
-					else if (icc->conCommand[0])
-					{ //the vm call says to execute this command in place
-						Cbuf_AddText( icc->conCommand );
-						Cbuf_AddText ("\n");
-					}
-				}
-				else
-				{ //otherwise just add it
-					Cbuf_AddText (kb);
-					Cbuf_AddText ("\n");
-				}
+				Cbuf_AddText (kb);
+				Cbuf_AddText ("\n");
 			}
 		}
 	}
@@ -1698,8 +1401,8 @@ void Key_ClearStates (void)
 
 	kg.anykeydown = qfalse;
 
-	for ( i=0 ; i < MAX_KEYS ; i++ ) {
-		if (i == A_SCROLLLOCK || i == A_NUMLOCK || i == A_CAPSLOCK)
+	for ( i=0 ; i < K_LAST_KEY ; i++ ) {
+		if (i == K_KP_NUMLOCK || i == K_CAPSLOCK)
 			continue;
 
 		if ( kg.keys[i].down ) {
diff --git a/codemp/client/cl_main.cpp b/codemp/client/cl_main.cpp
index e82bd00..544548b 100644
