--- a/codemp/game/g_xcvar.h
+++ b/codemp/game/g_xcvar.h
@@ -38,7 +38,7 @@ XCVAR_DEF( d_saberSPStyleDamage,		"1",			NULL,				CVAR_ARCHIVE,									qfalse )
 XCVAR_DEF( d_saberStanceDebug,			"0",			NULL,				CVAR_NONE,										qfalse )
 XCVAR_DEF( d_siegeSeekerNPC,			"0",			NULL,				CVAR_CHEAT,										qtrue )
 XCVAR_DEF( dedicated,					"0",			NULL,				CVAR_NONE,										qfalse )
-XCVAR_DEF( developer,					"0",			NULL,				CVAR_NONE,										qfalse )
+XCVAR_DEF( developer,					"1",			NULL,				CVAR_NONE,										qfalse )
 XCVAR_DEF( dmflags,						"0",			NULL,				CVAR_SERVERINFO|CVAR_ARCHIVE,					qtrue )
 XCVAR_DEF( duel_fraglimit,				"10",			NULL,				CVAR_SERVERINFO|CVAR_ARCHIVE|CVAR_NORESTART,	qtrue )
 XCVAR_DEF( fraglimit,					"20",			NULL,				CVAR_SERVERINFO|CVAR_ARCHIVE|CVAR_NORESTART,	qtrue )
diff --git a/codemp/game/game.vcxproj b/codemp/game/game.vcxproj
index d019bf5..4e5d40b 100644
