--- a/codemp/server/sv_init.cpp
+++ b/codemp/server/sv_init.cpp
@@ -12,8 +12,6 @@ Ghoul2 Insert Start
 #include "qcommon/MiniHeap.h"
 #endif
 
-#include "qcommon/stringed_ingame.h"
-
 /*
 ===============
 SV_SendConfigstring
@@ -563,10 +561,6 @@ Ghoul2 Insert End
 		sv.configstrings[i] = CopyString("");
 	}
 
-	//rww - RAGDOLL_BEGIN
-	re.G2API_SetTime(sv.time,0);
-	//rww - RAGDOLL_END
-
 	// make sure we are not paused
 	Cvar_Set("cl_paused", "0");
 
@@ -605,17 +599,11 @@ Ghoul2 Insert End
 
 	// run a few frames to allow everything to settle
 	for ( i = 0 ;i < 3 ; i++ ) {
-		//rww - RAGDOLL_BEGIN
-		re.G2API_SetTime(sv.time,0);
-		//rww - RAGDOLL_END
 		VM_Call( gvm, GAME_RUN_FRAME, sv.time );
 		SV_BotFrame( sv.time );
 		sv.time += 100;
 		svs.time += 100;
 	}
-	//rww - RAGDOLL_BEGIN
-	re.G2API_SetTime(sv.time,0);
-	//rww - RAGDOLL_END
 
 	// create a baseline for more efficient communications
 	SV_CreateBaseline ();
@@ -672,9 +660,6 @@ Ghoul2 Insert End
 	SV_BotFrame( sv.time );
 	sv.time += 100;
 	svs.time += 100;
-	//rww - RAGDOLL_BEGIN
-	re.G2API_SetTime(sv.time,0);
-	//rww - RAGDOLL_END
 
 	if ( sv_pure->integer ) {
 		// the server sends these to the clients so they will only
@@ -909,7 +894,6 @@ void SV_Init (void) {
 	// Get these to establish them and to make sure they have a default before the menus decide to stomp them.
 	Cvar_Get ("g_maxHolocronCarry", "3", CVAR_SERVERINFO);
 	Cvar_Get ("g_privateDuel", "1", CVAR_SERVERINFO );
-	Cvar_Get ("g_saberLocking", "1", CVAR_SERVERINFO );
 	Cvar_Get ("g_maxForceRank", "7", CVAR_SERVERINFO );
 	Cvar_Get ("duel_fraglimit", "10", CVAR_SERVERINFO);
 	Cvar_Get ("g_forceBasedTeams", "0", CVAR_SERVERINFO);
@@ -950,8 +934,7 @@ void SV_Init (void) {
 
 	sv_allowDownload = Cvar_Get ("sv_allowDownload", "0", CVAR_SERVERINFO);
 	sv_master[0] = Cvar_Get ("sv_master1", MASTER_SERVER_NAME, 0 );
-	sv_master[1] = Cvar_Get("sv_master2", JKHUB_MASTER_SERVER_NAME, 0);
-	for(int index = 2; index < MAX_MASTER_SERVERS; index++)
+	for(int index = 1; index < MAX_MASTER_SERVERS; index++)
 		sv_master[index] = Cvar_Get(va("sv_master%d", index + 1), "", CVAR_ARCHIVE);
 	sv_reconnectlimit = Cvar_Get ("sv_reconnectlimit", "3", 0);
 	sv_showghoultraces = Cvar_Get ("sv_showghoultraces", "0", 0);
diff --git a/codemp/server/sv_main.cpp b/codemp/server/sv_main.cpp
index 20ec9ce..79a68c3 100644
