--- a/codemp/server/sv_main.cpp
+++ b/codemp/server/sv_main.cpp
@@ -567,7 +567,7 @@ if a user is interested in a server to do a full status
 ================
 */
 void SVC_Info( netadr_t from ) {
-	int		i, count, humans, wDisable;
+	int		i, count, humans;
 	char	*gamedir;
 	char	infostring[MAX_INFO_STRING];
 
@@ -630,19 +630,8 @@ void SVC_Info( netadr_t from ) {
 	Info_SetValueForKey(infostring, "g_humanplayers", va("%i", humans));
 	Info_SetValueForKey( infostring, "sv_maxclients", 
 		va("%i", sv_maxclients->integer - sv_privateClients->integer ) );
-	Info_SetValueForKey( infostring, "gametype", va("%i", sv_gametype->integer ) );
+	Info_SetValueForKey( infostring, "gametype", sv_gametype->string );
 	Info_SetValueForKey( infostring, "needpass", va("%i", sv_needpass->integer ) );
-	Info_SetValueForKey( infostring, "truejedi", va("%i", Cvar_VariableIntegerValue( "g_jediVmerc" ) ) );
-	if ( sv_gametype->integer == GT_DUEL || sv_gametype->integer == GT_POWERDUEL )
-	{
-		wDisable = Cvar_VariableIntegerValue( "g_duelWeaponDisable" );
-	}
-	else
-	{
-		wDisable = Cvar_VariableIntegerValue( "g_weaponDisable" );
-	}
-	Info_SetValueForKey( infostring, "wdisable", va("%i", wDisable ) );
-	Info_SetValueForKey( infostring, "fdisable", va("%i", Cvar_VariableIntegerValue( "g_forcePowerDisable" ) ) );
 	//Info_SetValueForKey( infostring, "pure", va("%i", sv_pure->integer ) );
 
 	if( sv_minPing->integer ) {
@@ -1130,10 +1119,6 @@ void SV_Frame( int msec ) {
 		VM_Call( gvm, GAME_RUN_FRAME, sv.time );
 	}
 
-	//rww - RAGDOLL_BEGIN
-	re.G2API_SetTime(sv.time,0);
-	//rww - RAGDOLL_END
-
 	if ( com_speeds->integer ) {
 		time_game = Sys_Milliseconds () - startTime;
 	}
diff --git a/codemp/server/sv_snapshot.cpp b/codemp/server/sv_snapshot.cpp
index 852ac2e..f22dba9 100644
